# 🏨 **Hotel Cancel Culture: Who Will Cancel?** 🏨

**Who?**
>* 🏢 **Revenue Management (RM) teams** for hotel groups (corporate, franchise)
>
>
>* 🏨 On-site GMs, Sales, and Ops teams

**Why?**
>* 💰 **Revenue Management:** 
>  * Revenue optimization: Right price, right time, right customer
>    * Dynamic pricing
>    * Distribution channels
>    * Pricing per room type
>
>
>* 🤝 **Sales:**
>  * Group sales (pickup/wash)
>  * BT (performance/company for both GPP and LNR rates)
>
>
>* 🛌 **Rooms Ops:**
>  * Forecasting occupancy, arrivals, departures, stay-overs, same-day booking demand, and probability of guest relocation in the case of oversell.
>  * Determining staff schedules and periods of high demand
>
>
>* 🍰 ☕ **Food and Beverage:**
>  * Ordering food/supplies overall
>  * Scheduling staff
>  * Determining busy times (breakfast, lunch, dinner)
>    * Staffing, specific food/supplies

**What?**
>* 🧾 Dataset comprised of... 
>  * 32 different features
>    * Detailed explanation of features (and sub-categories, when appropriate) available in Readme
>  * Nearly 120,000 reservation records
>  * Source cited in Readme

❌ **How?**
>* Which models/methods? 
>* Data prep and feature engineering

**Goal:** determine whether or not a reservation would cancel given different details of a guest and their reservation.

# 📌 **To-Do**

>* Update each section with ***working*** code
>* Include references/details for relevant packages
---

- [ ] [TD1](#td1)
- [ ] [TD2](#td2)
- [ ] [TD3](#td3)
- [ ] [todo4](#td4)
- [ ] [todo5](#td5)
- [ ] [todo6](#td6)
- [ ] [todo7](#td7)

# 📂 **Imports**

In [13]:
# Data Handling
import pandas as pd
import numpy as np
from scipy import stats

# Visualizations
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib as mpl

# Modeling - SKLearn
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import StandardScaler, OneHotEncoder, FunctionTransformer
from sklearn.model_selection import train_test_split, cross_val_score
from sklearn.impute import SimpleImputer
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.dummy import DummyRegressor

# %load_ext autoreload
# %autoreload 2

import clf_functions.functions as cf

# from sklearn.naive_bayes import MultinomialNB # for naive bayes model

# Settings
%matplotlib inline
plt.style.use('seaborn-talk')
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', lambda x: f'{x:,.2f}')
pd.set_option('max_rows', 100)

from sklearn import set_config
set_config(display='diagram')

# 📖 **Read Data**

In [58]:
## Reading data and saving to a DataFrame

data_link = 'http://data.insideairbnb.com/united-states/dc/washington-dc/2021-07-10/data/listings.csv.gz'

data = pd.read_csv(data_link)

In [59]:
data.head(5)

id                        listing_url       scrape_id last_scraped  \
0  3686  https://www.airbnb.com/rooms/3686  20210710190002   2021-07-11   
1  3943  https://www.airbnb.com/rooms/3943  20210710190002   2021-07-11   
2  4529  https://www.airbnb.com/rooms/4529  20210710190002   2021-07-11   
3  4967  https://www.airbnb.com/rooms/4967  20210710190002   2021-07-11   
4  5589  https://www.airbnb.com/rooms/5589  20210710190002   2021-07-11   

                               name  \
0                   Vita's Hideaway   
1  Historic Rowhouse Near Monuments   
2         Bertina's  House Part One   
3                    DC, Near Metro   
4          Cozy apt in Adams Morgan   

                                         description  \
0  IMPORTANT NOTES<br />* Carefully read and be s...   
1  Please contact us before booking to make sure ...   
2  This is  large private  bedroom with plenty of...   
3  <b>The space</b><br />Hello, my name is Seveer...   
4  This is a 1 br (bedroom + living room in Adams...   

                               neighborhood_overview  \
0  We love that our neighborhood is up and coming...   
1  This rowhouse is centrally located in the hear...   
2  Very quiet neighborhood and it is easy accessi...   
3                                                NaN   
4  Adams Morgan spills over with hipsters, salsa ...   

                                         picture_url  host_id  \
0  https://a0.muscache.com/pictures/61e02c7e-3d66...     4645   
1  https://a0.muscache.com/pictures/432713/fab7dd...     5059   
2  https://a0.muscache.com/pictures/86072003/6709...     5803   
3  https://a0.muscache.com/pictures/2439810/bb320...     7086   
4  https://a0.muscache.com/pictures/207249/9f1df8...     6527   

                                 host_url        host_name  host_since  \
0  https://www.airbnb.com/users/show/4645             Vita  2008-11-26   
1  https://www.airbnb.com/users/show/5059             Vasa  2008-12-12   
2  https://www.airbnb.com/users/show/5803  Bertina'S House  2008-12-30   
3  https://www.airbnb.com/users/show/7086           Seveer  2009-01-26   
4  https://www.airbnb.com/users/show/6527              Ami  2009-01-13   

                                       host_location  \
0  Washington D.C., District of Columbia, United ...   
1    Washington, District of Columbia, United States   
2    Washington, District of Columbia, United States   
3  Washington D.C., District of Columbia, United ...   
4  Washington D.C., District of Columbia, United ...   

                                          host_about  host_response_time  \
0  I am a literary scholar, teacher, poet, vegan ...        within a day   
1  I have been living and working in DC for the l...  within a few hours   
2  I am an easy going, laid back person who loves...                 NaN   
3  I am fun, honest and very easy going and trave...  within a few hours   
4  I am an environmentalist, and I own and operat...  within a few hours   

  host_response_rate host_acceptance_rate host_is_superhost  \
0                80%                  75%                 f   
1               100%                  29%                 f   
2                NaN                  NaN                 f   
3               100%                  78%                 t   
4               100%                  17%                 f   

                                  host_thumbnail_url  \
0  https://a0.muscache.com/im/users/4645/profile_...   
1  https://a0.muscache.com/im/pictures/user/8ec69...   
2  https://a0.muscache.com/im/users/5803/profile_...   
3  https://a0.muscache.com/im/pictures/user/6efb4...   
4  https://a0.muscache.com/im/users/6527/profile_...   

                                    host_picture_url host_neighbourhood  \
0  https://a0.muscache.com/im/users/4645/profile_...          Anacostia   
1  https://a0.muscache.com/im/pictures/user/8ec69...          Eckington   
2  https://a0.muscache.com/im/users/5803/profile_...   Eastland Gardens   

In [60]:
data.shape

(8033, 74)

> The initial reading of the dataset shows there are 74 features and 8,033 entries. A quick glance at the `.head()` gives a sample of the entries, showing that some of the features are not relevant to my analysis.
>
> I will drop the following features now as they are metadata related to the webscraping required to provide the data (and would not have any meaningful impact on my analysis):
>
>* listing_url
>
>
>* scrape_id
>
>
>* last_scraped
>
>
>* picture_url
>
>
>* host_url
>
>
>* host_thumbnail_url
>
>
>* host_picture_url
>
>
>* calendar_last_scraped

In [62]:
df = data.drop(columns= ['listing_url', 'scrape_id', 'last_scraped', 'picture_url',
                         'host_url', 'host_thumbnail_url','host_picture_url',
                         'calendar_last_scraped']).copy()

In [63]:
## Creating features/target for dataset
target = 'review_scores_rating'

X = df.drop(columns = target).copy()
y = df[target].copy()

In [64]:
## Confirming same number of rows
X.shape[0] == y.shape[0]

True

In [65]:
## Splitting to prevent data leakage
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

# 🧼 **Starting Cleaning and EDA**

In [70]:
## Checking for class imbalances
y_train.describe()

count   4,406.00
mean        4.69
std         0.69
min         0.00
25%         4.68
50%         4.86
75%         5.00
max         5.00
Name: review_scores_rating, dtype: float64

> The target feature, `'review_scores_rating'`, is currently a range of values from 0 to 5, with 75% of the scores being 4.68 or above. In order to use these values, I would need to bin them into categories based on ranges of values for analysis.

From original article:

> ❌

In [17]:
report = cf.report_df(X_train)
report

(6024, 73)


datatypes  num_unique  null_sum  \
id                                               int64        6024         0   
listing_url                                     object        6024         0   
scrape_id                                        int64           1         0   
last_scraped                                    object           3         0   
name                                            object        5700         1   
description                                     object        5401       122   
neighborhood_overview                           object        3268      2173   
picture_url                                     object        5779         0   
host_id                                          int64        3762         0   
host_url                                        object        3762         0   
host_name                                       object        2063        92   
host_since                                      object        2132        92   
host_location                                   object         316       104   
host_about                                      object        2153      2378   
host_response_time                              object           4      1968   
host_response_rate                              object          54      1968   
host_acceptance_rate                            object          82      1670   
host_is_superhost                               object           2        92   
host_thumbnail_url                              object        3744        92   
host_picture_url                                object        3744        92   
host_neighbourhood                              object         179       557   
host_listings_count                            float64          57        92   
host_total_listings_count                      float64          57        92   
host_verifications                              object         267         0   
host_has_profile_pic                            object           2        92   
host_identity_verified                          object           2        92   
neighbourhood                                   object          18      2173   
neighbourhood_cleansed                          object          39         0   
neighbourhood_group_cleansed                   float64           0      6024   
latitude                                       float64        4230         0   
longitude                                      float64        4478         0   
property_type                                   object          46         0   
room_type                                       object           4         0   
accommodates                                     int64          17         0   
bathrooms                                      float64           0      6024   
bathrooms_text                                  object          29         8   
bedrooms                                       float64           9       628   
beds                                           float64          17       226   
amenities                                       object        5319         0   
price                                           object         514         0   
minimum_nights                                   int64          58         0   
maximum_nights                                   int64         131         0   
minimum_minimum_nights                           int64          60         0   
maximum_minimum_nights                           int64          62         0   
minimum_maximum_nights                           int64         107         0   
maximum_maximum_nights                           int64         108         0   
minimum_nights_avg_ntm                         float64         195         0   
maximum_nights_avg_ntm                         float64         146         0   
calendar_updated                               float64           0      6024   
has_availability                             

In [23]:
# df2.style.bar(subset=['A', 'B'], color='#d65f5f')

report.style.bar(subset=['null_pct'], color='#d65f5f')#, vmin=0, vmax=1)

In [33]:
from ipywidgets import interact

In [55]:
@interact
def sort_report(Column=['num_unique', 'null_pct'], ascending = False):
    display(report.sort_values(Column, ascending=ascending).style.bar(subset=[Column], color='#d65f5f'))

interactive(children=(Dropdown(description='Column', options=('num_unique', 'null_pct'), value='num_unique'), …

In [54]:
import fsds as fs

In [47]:
fs.ihelp_menu([fs.ihelp_menu, sort_report])

GridBox(box_style='warning', children=(VBox(children=(Label(value='iHelp Menu: View Help and/or Source Code'),…

Output()

In [32]:
sort_report('null_sum', ascending=True)

## ❓ **Null Values and How to Address:**

>* **Children:** fill in missing values with zero (75% of the data is zero, making it an easy and accurate choice)
>
>
>* **Country, Agent:** For these three sections, the documentation for the dataset indicates that "NULL" or NaN values are not truly *missing* from the dataset, but that the data source did not have any values for those reservations.
>  * **To address:** I will impute the string, "missing," as a placeholder to keep these features in the dataset.
>
>
>* **Company:** Since this feature is 94% empty values, it would have an insignificant impact on the predictions of whether or not a guest will cancel their reservation.
>  * **To address:** I will drop the column as part of the 

In [9]:
X_train.head()

hotel  lead_time  arrival_date_year arrival_date_month  \
118418    City Hotel        208               2017             August   
66950     City Hotel        110               2017              April   
52240     City Hotel        330               2016               June   
3835    Resort Hotel          3               2016            January   
3330    Resort Hotel         66               2015           December   

        arrival_date_week_number  arrival_date_day_of_month  \
118418                        33                         17   
66950                         17                         26   
52240                         23                          2   
3835                           4                         21   
3330                          49                          5   

        stays_in_weekend_nights  stays_in_week_nights  adults  children  \
118418                        0                     3       2      0.00   
66950                         1                     4       2      0.00   
52240                         0                     2       2      0.00   
3835                          0                     1       1      0.00   
3330                          1                     1       2      0.00   

        babies meal country market_segment distribution_channel  \
118418       0   BB     DEU      Online TA                TA/TO   
66950        0   SC     NOR      Online TA                TA/TO   
52240        0   BB     PRT         Groups                TA/TO   
3835         0   BB     PRT      Corporate            Corporate   
3330         0   FB     PRT         Groups                TA/TO   

        is_repeated_guest  previous_cancellations  \
118418                  0                       0   
66950                   0                       0   
52240                   0                       0   
3835                    0                       0   
3330                    0                       0   

        previous_bookings_not_canceled reserved_room_type assigned_room_type  \
118418                               0                  A                  A   
66950                                0                  A                  A   
52240                                0                  A                  A   
3835                                 0                  A                  E   
3330                                 0                  A                  A   

        booking_changes deposit_type  agent  company  days_in_waiting_list  \
118418                0   No Deposit   9.00      nan                     0   
66950                 0   No Deposit   9.00      nan                     0   
52240                 0   Non Refund   1.00      nan                     0   
3835                  0   No Deposit    nan    47.00                     0   
3330                  0   No Deposit 155.00      nan                     0   

          customer_type    adr  required_car_parking_spaces  \
118418        Transient 131.70                            1   
66950         Transient 100.80                            0   
52240         Transient  62.00                            0   
3835          Transient  30.00                            0   
3330    Transient-Party  64.00                            0   

        total_of_special_requests reservation_status reservation_status_date  
118418                          0          Check-Out              2017-08-20  
66950                           0           Canceled              2017-01-06  
52240                           0           Canceled              2015-10-21  
3835                            0           Canceled              2016-01-18  
3330                            0           Canceled              2015-12-02

In [10]:
X_train['meal'].value_counts()

BB           69242
HB           10794
SC            8027
Undefined      888
FB             591
Name: meal, dtype: int64

In [11]:
und_count = X_train.isin(['Undefined']).sum()
# und_count

In [12]:
und_count[und_count > 0]

meal                    888
market_segment            2
distribution_channel      4
dtype: int64

#### Old code - saving just in case

In [13]:
## Unsure what this code was for - something about the "Undefined" values? 

# for col in X_train.columns:
#     if X_train[col].dtype == 'O':
#         print(X_train[col].value_counts(),'\n\n')

In [14]:
# for col in X_train.columns:
#     if 'Undefined' in X_train[col]:
#         print(col)

In [15]:
# 'undefined' in X_train["meal"]

In [16]:
# X_train['distribution_channel'].unique()

In [17]:
# #3 Identifying which columns indicate missing values based on .isna() method

# nan_cols = []

# for column in X_train.columns:
#     if X_train[column].isna().sum() > 0:
#         nan_cols.append(column)
#         print(column,'\n\n',X_train[column].value_counts(dropna=False),'\n\n')
        
# nan_cols

In [18]:
# for column in X_train.columns:
#     if X_train[column].isna().sum() > 0:
# #         nan_cols.append(column)
#         print(column,'\n\n',X_train[column].isna().sum(),'\n\n')

In [19]:
# X_train['country'].isna().sum()

In [20]:
# X_train[X_train['country'].isna()]

# 🚿 **Preprocessing Pipeline**

In [23]:
cat_cols = ['hotel', 'meal','arrival_date_month', 'country', 'market_segment',
            'distribution_channel','is_repeated_guest','reserved_room_type',
            'assigned_room_type','deposit_type', 'agent',
            'customer_type','reservation_status']

cont_cols = [col for col in X_train.drop(['reservation_status_date','company'],axis=1).columns if col not in cat_cols]

cont_cols

['lead_time',
 'arrival_date_year',
 'arrival_date_week_number',
 'arrival_date_day_of_month',
 'stays_in_weekend_nights',
 'stays_in_week_nights',
 'adults',
 'children',
 'babies',
 'previous_cancellations',
 'previous_bookings_not_canceled',
 'booking_changes',
 'days_in_waiting_list',
 'adr',
 'required_car_parking_spaces',
 'total_of_special_requests']

In [24]:
X_train[cat_cols] = X_train[cat_cols].astype(str)

C:\Users\bmcca\anaconda3\envs\learn-env-bmc\lib\site-packages\pandas\core\frame.py:3065: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]


In [25]:
X_test[cat_cols] = X_test[cat_cols].astype(str)

In [26]:
## Creating ColumnTransformer and sub-transformers for imputation and encoding

# Filling missing "Children"
zero_transformer = SimpleImputer(strategy='constant', fill_value=0)

##  
missing_transformer = SimpleImputer(strategy='constant', fill_value='missing')

## Encoding categoricals - handling errors to prevent issues w/ test set
categorical_transformer = OneHotEncoder(handle_unknown='ignore', sparse=False)

cat_pipe = Pipeline(steps=[('imputer', missing_transformer),
                      ('ohe', categorical_transformer)])

cont_pipe = Pipeline(steps=[('imputer', zero_transformer),
                           ('scaler', StandardScaler())])

## Instantiating the ColumnTransformer and including all transformers
preprocessor = ColumnTransformer(
    transformers=[('conts', cont_pipe, cont_cols),
                  ('cats', cat_pipe, cat_cols)])

preprocessor

ColumnTransformer(transformers=[('conts',
                                 Pipeline(steps=[('imputer',
                                                  SimpleImputer(fill_value=0,
                                                                strategy='constant')),
                                                 ('scaler', StandardScaler())]),
                                 ['lead_time', 'arrival_date_year',
                                  'arrival_date_week_number',
                                  'arrival_date_day_of_month',
                                  'stays_in_weekend_nights',
                                  'stays_in_week_nights', 'adults', 'children',
                                  'babies', 'previous_cancellations',
                                  'previous_bo...
                                 Pipeline(steps=[('imputer',
                                                  SimpleImputer(fill_value='missing',
                                                                strategy='constant')),
                                                 ('ohe',
                                                  OneHotEncoder(handle_unknown='ignore',
                                                                sparse=False))]),
                                 ['hotel', 'meal', 'arrival_date_month',
                                  'country', 'market_segment',
                                  'distribution_channel', 'is_repeated_guest',
                                  'reserved_room_type', 'assigned_room_type',
                                  'deposit_type', 'agent', 'customer_type',
                                  'reservation_status'])])

In [27]:
preprocessor.fit(X_train)

## Getting feature names from OHE
ohe_cat_names = preprocessor.named_transformers_['cats'].named_steps['ohe'].get_feature_names(cat_cols)

In [28]:
## Generating list for column index
final_cols = [*cont_cols, *ohe_cat_names]

## Fit and transform the data via the ColumnTransformer
X_train_tf = preprocessor.transform(X_train)
X_train_tf_df = pd.DataFrame(X_train_tf, columns=final_cols, index=X_train.index)

## Transforming the test set and saving
X_test_tf = preprocessor.transform(X_test)
X_test_tf_df = pd.DataFrame(X_test_tf, columns=final_cols, index=X_test.index)

display(X_train_tf_df.head(5),X_test_tf_df.head(5))

lead_time  arrival_date_year  arrival_date_week_number  \
118418       0.97               1.19                      0.43   
66950        0.06               1.19                     -0.75   
52240        2.12              -0.22                     -0.31   
3835        -0.94              -0.22                     -1.70   
3330        -0.35              -1.63                      1.61   

        arrival_date_day_of_month  stays_in_weekend_nights  \
118418                       0.14                    -0.93   
66950                        1.16                     0.07   
52240                       -1.57                    -0.93   
3835                         0.59                    -0.93   
3330                        -1.23                     0.07   

        stays_in_week_nights  adults  children  babies  \
118418                  0.26    0.25     -0.26   -0.08   
66950                   0.78    0.25     -0.26   -0.08   
52240                  -0.26    0.25     -0.26   -0.08   
3835                   -0.78   -1.48     -0.26   -0.08   
3330                   -0.78    0.25     -0.26   -0.08   

        previous_cancellations  previous_bookings_not_canceled  \
118418                   -0.10                           -0.09   
66950                    -0.10                           -0.09   
52240                    -0.10                           -0.09   
3835                     -0.10                           -0.09   
3330                     -0.10                           -0.09   

        booking_changes  days_in_waiting_list   adr  \
118418            -0.34                 -0.13  0.58   
66950             -0.34                 -0.13 -0.02   
52240             -0.34                 -0.13 -0.78   
3835              -0.34                 -0.13 -1.40   
3330              -0.34                 -0.13 -0.74   

        required_car_parking_spaces  total_of_special_requests  \
118418                         3.81                      -0.72   
66950                         -0.25                      -0.72   
52240                         -0.25                      -0.72   
3835                          -0.25                      -0.72   
3330                          -0.25                      -0.72   

        hotel_City Hotel  hotel_Resort Hotel  meal_BB  meal_FB  meal_HB  \
118418              1.00                0.00     1.00     0.00     0.00   
66950               1.00                0.00     0.00     0.00     0.00   
52240               1.00                0.00     1.00     0.00     0.00   
3835                0.00                1.00     1.00     0.00     0.00   
3330                0.00                1.00     0.00     1.00     0.00   

        meal_SC  meal_Undefined  arrival_date_month_April  \
118418     0.00            0.00                      0.00   
66950      1.00            0.00                      1.00   
52240      0.00            0.00                      0.00   
3835       0.00            0.00                      0.00   
3330       0.00            0.00                      0.00   

        arrival_date_month_August  arrival_date_month_December  \
118418                       1.00                         0.00   
66950                        0.00                         0.00   
52240                        0.00                         0.00   
3835                         0.00                         0.00   
3330                         0.00                         1.00   

        arrival_date_month_February  arrival_date_month_January  \
118418                         0.00                        0.00   
66950                          0.00                        0.00   
52240                          0.00                        0.00   
3835                           0.00                        1.00   
3330                           0.00                        0.00   

        arrival_date_month_July  arrival_date_month_June  \
118418                     0.00                     0.00   
66950                     

lead_time  arrival_date_year  arrival_date_week_number  \
30946        0.93              -0.22                      1.61   
40207       -0.20              -1.63                      0.14   
103708      -0.74              -0.22                      1.90   
85144       -0.96              -0.22                     -1.19   
109991      -0.32               1.19                     -0.82   

        arrival_date_day_of_month  stays_in_weekend_nights  \
30946                       -1.57                     1.07   
40207                        0.02                    -0.93   
103708                       1.28                    -0.93   
85144                       -0.77                    -0.93   
109991                       0.02                     1.07   

        stays_in_week_nights  adults  children  babies  \
30946                   1.30    0.25     -0.26   -0.08   
40207                   0.26    0.25     -0.26   -0.08   
103708                  0.26    1.98     -0.26   -0.08   
85144                  -0.78   -1.48     -0.26   -0.08   
109991                 -0.26    0.25     -0.26   -0.08   

        previous_cancellations  previous_bookings_not_canceled  \
30946                    -0.10                           -0.09   
40207                    -0.10                           -0.09   
103708                   -0.10                           -0.09   
85144                    -0.10                           -0.09   
109991                   -0.10                           -0.09   

        booking_changes  days_in_waiting_list   adr  \
30946              5.79                 -0.13 -0.68   
40207             -0.34                 -0.13 -0.49   
103708             2.73                 -0.13 -0.81   
85144             -0.34                 -0.13 -0.13   
109991            -0.34                 -0.13  0.12   

        required_car_parking_spaces  total_of_special_requests  \
30946                         -0.25                      -0.72   
40207                         -0.25                      -0.72   
103708                        -0.25                       0.54   
85144                         -0.25                      -0.72   
109991                        -0.25                      -0.72   

        hotel_City Hotel  hotel_Resort Hotel  meal_BB  meal_FB  meal_HB  \
30946               0.00                1.00     1.00     0.00     0.00   
40207               1.00                0.00     1.00     0.00     0.00   
103708              1.00                0.00     1.00     0.00     0.00   
85144               1.00                0.00     1.00     0.00     0.00   
109991              1.00                0.00     0.00     0.00     0.00   

        meal_SC  meal_Undefined  arrival_date_month_April  \
30946      0.00            0.00                      0.00   
40207      0.00            0.00                      0.00   
103708     0.00            0.00                      0.00   
85144      0.00            0.00                      0.00   
109991     1.00            0.00                      1.00   

        arrival_date_month_August  arrival_date_month_December  \
30946                        0.00                         1.00   
40207                        0.00                         0.00   
103708                       0.00                         1.00   
85144                        0.00                         0.00   
109991                       0.00                         0.00   

        arrival_date_month_February  arrival_date_month_January  \
30946                          0.00                        0.00   
40207                          0.00                        0.00   
103708                         0.00                        0.00   
85144                          0.00                        0.00   
109991                         0.00                        0.00   

        arrival_date_month_July  arrival_date_month_June  \
30946                      0.00                     0.00   
40207                     

# 📝 Next Steps

* Process classification model - i.e. Logreg, KNN, DecisionTrees, etc.
* Evaluate results
* Determine if I need to redo pre-processing steps

# 🚿 Classification Pipeline